### DISCUSSION TITLE
SDK and Self Signed Certs Don't Work ?

### DISCUSSION THREAD

**Discussion by bensoer** [aws/aws-sdk-js-v3#3652] [Q&A]
hello, Im having issues getting a self signed cert to work with CloudFormationClient():

```
const cloudFormationClient = new CloudFormationClient({
      logger: console,
      credentials: provider,
      requestHandler: new NodeHttpHandler({
        httpAgent: new http.Agent({}),
        httpsAgent: new https.Agent({
          ca: fs.readFileSync(bundlePath),
          rejectUnauthorized: true,

        })
    }),
      region: 'us-east-1',
    })
```
Ive tried following docs here:

https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-registering-certs.html
https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/node-registering-certs.html
https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/node-configuring-proxies.html

But nothing seems to work. I get still errors of self signed certs in the chain regardless of what configuration I seem to use, it looks like the Client doesn't even pickup or use my changes. Notably, the httpOptions attribute does not exist as part of the config. So im lost on what to do here. Thanks for any help!

**Comment by ajredniwja** [comment]
@bensoer thanks for opening this issue, I was able to follow through [this](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/node-configuring-proxies.html) tutorial to configure proxies and I used `https-proxy-agent` package. Would you be able to share more information what error you are getting and what version of the SDK you are using.

**Reply by bensoer** [reply]
@ajredniwja I am using v3 of the javascript SDK. I've read the same in some places that it works in v2. Also, i'm not trying to setup a proxy, though I have looked at that documentation for ideas to try something different.

My goal is to be able to use self-signed certificates in the javascript v3 SDK :)
